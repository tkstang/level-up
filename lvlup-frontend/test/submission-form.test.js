import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import lvlupApp from '../src/reducers/index';
import SubmissionMainForm from '../src/components/Student/submissions/submission-form';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

test('SubmissionMainForm should render', () => {
  const store = mockStore({ lvlupApp });
  const sub = shallow(
    <SubmissionMainForm selectedChallenge={{ name: 'Feed the Hamid' }} handleSubmit={jest.fn} store={store} />,
  );
  expect(shallowToJson(sub)).toMatchSnapshot();
});
