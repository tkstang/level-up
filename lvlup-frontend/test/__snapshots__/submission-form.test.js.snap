exports[`test SubmissionMainForm should render 1`] = `
<div
  className="submission-container">
  <Container>
    <div
      className="challenge-submission">
      <h2
        className="header">
        Challenge Submission: Feed the Hamid
      </h2>
      <Segment
        inverted={true}>
        <Form
          as="form"
          className="challenge-submission-form"
          onSubmit={[Function]}>
          <FormField
            inline={true}>
            <Field
              component={[Function]}
              label="Submission Message"
              name="submission_message"
              placeholder="Enter Submission Message"
              type="text"
              validate={
                Array [
                  [Function],
                  [Function],
                ]
              } />
          </FormField>
          <FormField
            inline={true}>
            <Field
              component={[Function]}
              label="Attachment Link 1"
              name="submission_attachment_1"
              placeholder="Please add your attachment here"
              type="text" />
          </FormField>
          <FormField
            inline={true}>
            <Field
              component={[Function]}
              label="Attachment Link 2"
              name="submission_attachment_2"
              placeholder="Please add your attachment here"
              type="text" />
          </FormField>
          <FormField
            inline={true}>
            <Field
              component={[Function]}
              label="Attachment Link 3"
              name="submission_attachment_3"
              placeholder="Please add your attachment here"
              type="text" />
          </FormField>
          <FormField
            id="submit-image"
            inline={true}>
            <Field
              component={[Function]}
              label="Submission Image 1"
              name="submission_image_link_1"
              placeholder="Please choose a file below and copy link here"
              type="text" />
          </FormField>
          <div>
            <input
              name="submission_image_link_1"
              role="uploadcare-uploader"
              type="text" />
          </div>
          <FormField
            id="submit-image"
            inline={true}>
            <Field
              component={[Function]}
              label="Submission Image 2"
              name="submission_image_link_2"
              placeholder="Please choose a file below and copy link here"
              type="text" />
          </FormField>
          <div>
            <input
              name="submission_image_link_2"
              role="uploadcare-uploader"
              type="text" />
          </div>
          <FormField
            id="submit-image"
            inline={true}>
            <Field
              component={[Function]}
              label="Submission Image 3"
              name="submission_image_link_3"
              placeholder="Please choose a file below and copy link here"
              type="text" />
          </FormField>
          <div>
            <input
              name="submission_image_link_3"
              role="uploadcare-uploader"
              type="text" />
          </div>
          <FormButton
            as={[Function]}
            basic={true}
            color="orange"
            control={[Function]}
            floated="right">
            lvl^
          </FormButton>
        </Form>
      </Segment>
    </div>
  </Container>
</div>
`;
